diff --git a/drivers/gpio/gpio-rockchip.c b/drivers/gpio/gpio-rockchip.c
index 8469eb76813e..5a0841401a23 100644
--- a/drivers/gpio/gpio-rockchip.c
+++ b/drivers/gpio/gpio-rockchip.c
@@ -287,6 +287,10 @@ static int rockchip_gpio_set_config(struct gpio_chip *gc, unsigned int offset,
 	unsigned int debounce = pinconf_to_config_argument(config);
 
 	switch (param) {
+	case PIN_CONFIG_BIAS_DISABLE:
+	case PIN_CONFIG_BIAS_PULL_UP:
+	case PIN_CONFIG_BIAS_PULL_DOWN:
+		return gpiochip_generic_config(gc, offset, config);
 	case PIN_CONFIG_INPUT_DEBOUNCE:
 		rockchip_gpio_set_debounce(gc, offset, debounce);
 		/*
