diff --git a/drivers/soc/rockchip/pm_domains.c b/drivers/soc/rockchip/pm_domains.c
index 8bb2a5904f83..a4bd4dd1f028 100644
--- a/drivers/soc/rockchip/pm_domains.c
+++ b/drivers/soc/rockchip/pm_domains.c
@@ -876,7 +876,7 @@ static int rockchip_pmu_domain_mem_reset(struct rockchip_pm_domain *pd)
 		goto error;
 	}
 
-	udelay(60);
+	udelay(600);
 
 	regmap_write(pmu->regmap, pmu->info->mem_pwr_offset + pd->info->pwr_offset,
 		     (pd->info->pwr_mask | pd->info->pwr_w_mask));
